src/IIIFPresentation/API/Converters/CollectionConverter.cs

```diff
@@ -24,60 +22,78 @@ public static Collection ToHierarchicalCollection(this Models.Database.Collectio
             }).ToList<ICollectionItem>() : new List<ICollectionItem>()
         };
 
-        return stuff;
+        return collection;
     }
-    
-    public static FlatCollection ToFlatCollection(this Models.Database.Collections.Collection dbAsset, UrlRoots urlRoots, int pageSize, 
-        IQueryable<Models.Database.Collections.Collection>? items)
+
+    public static FlatCollection ToFlatCollection(this Models.Database.Collections.Collection dbAsset,
+        UrlRoots urlRoots, int pageSize, int currentPage, int totalItems,
+        List<Models.Database.Collections.Collection>? items, string? orderQueryParam = null)
     {
-        var itemCount = items?.Count() ?? 0;
+        var totalPages = (int)Math.Ceiling(totalItems == 0 ? 1 : (double)totalItems / pageSize);
+
+        var id = $"{urlRoots.BaseUrl}/{dbAsset.CustomerId}/collections/{dbAsset.Id}";
+
+        var orderQueryParamConverted = string.IsNullOrEmpty(orderQueryParam) ? string.Empty : $"&{orderQueryParam}";
         
         return new FlatCollection()
         {
-            Id = $"{urlRoots.BaseUrl}/{dbAsset.CustomerId}/collections/{dbAsset.Id}",
-            Context = new List<string> 
-                {
-                    "http://iiif.io/api/presentation/3/context.json", 
-                    "http://tbc.org/iiif-repository/1/context.json" 
-                },
+            Id = id,
+            Context = new List<string>
+            {
+                "http://iiif.io/api/presentation/3/context.json",
+                "http://tbc.org/iiif-repository/1/context.json"
```

Comment:
The IIIF Presentation context must come last, see https://iiif.io/api/presentation/3.0/#46-linked-data-context-and-extensions (when this is using `Collection` we can use the `.EnsureXX()` method to enforce this)